java/maven/src/org/netbeans/modules/maven/api/PluginPropertyUtils.java

```diff
@@ -652,21 +657,48 @@ public List<Dependency> build(Xpp3Dom configRoot, ExpressionEvaluator eval) {
                     item.setGroupId(g.getValue());
                     item.setLocation(PROP_GROUP_ID, (InputLocation)g.getInputLocation());
                 }
+
+                Dependency dependencyFromDependencyManagement = null;
+                if(dependencyManagement != null && dependencyManagement.getDependencies() != null) {
+                    dependencyFromDependencyManagement =
+                            dependencyManagement
+                            .getDependencies()
+                            .stream()
+                            .filter(d -> {
+                                return Objects.equals(item.getGroupId(), d.getGroupId())
+                                        && Objects.equals(item.getArtifactId(), d.getArtifactId());
+                            })
+                            .findFirst()
+                            .orElse(null);
+                }
+
                 if (v != null) {
                     item.setVersion(v.getValue());
                     item.setLocation(PROP_VERSION, (InputLocation)v.getInputLocation());
+                } else if (dependencyFromDependencyManagement != null) {
+                    item.setVersion(dependencyFromDependencyManagement.getVersion());
+                    item.setLocation(PROP_VERSION, dependencyFromDependencyManagement.getLocation(PROP_VERSION));
                 }
                 if (c != null) {
                     item.setClassifier(c.getValue());
                     item.setLocation(PROP_CLASSIFIER, (InputLocation)c.getInputLocation());
+                } else if (dependencyFromDependencyManagement != null) {
```

Comment:
I am not sure about this: the `classifier` tags a differnt flavour of an artifact. So IMHO the dependency mgmt tagged with a classifier does not apply/manage a project dependency without any classifier (empty classifier). 

IMHO group:artifact:classifier:type of the dependency must match the dependency mgmt decl in order to apply the version from the dependency management.

So if the dependency mgmt declares a classifier and the artifact does not, or the dependency mgmt declares a type different from the assumed artifact type, the management does not apply.

Reply:
You are right. I had a different use-case in mind, but that is not covered:

https://github.com/apache/maven-compiler-plugin/blob/46b207b6fb4df8fdbc1188896e5245047324d925/src/main/java/org/apache/maven/plugin/compiler/DependencyCoordinate.java#L118-L126

This might be handled differently in other plugins, but for now I adjusted the code accordingly. Pushed an update.

Reply:
@sdedic would you mind having another look?

Reply:
Apologies for the delay.